server/clearbit.js

```diff
@@ -95,10 +97,44 @@ export default class Clearbit {
 
     this.metric("saveUser");
 
-    return this.hull
-      .as(ident)
-      .traits(traits)
-      .then(() => { return { user, person }; });
+    if (this.settings.handle_accounts) {
+      const all_traits = { user: {}, account: {} };
+
+      _.map(traits, (v, k) => {
+        const [group, trait] = k.split("/");
+        if (group === "clearbit_company") {
+          all_traits.account[`clearbit/${trait}`] = v;
+        } else {
+          all_traits.user[k] = v;
+        }
+      }, {});
+
+      const domain = all_traits.account["clearbit/domain"];
+
+      // Set top level traits
+      const top_level_traits = {
+        "name": "name",
+        "domain": "domain",
+      };
+      _.forIn(top_level_traits, (clearbit_name, top_level_name) => {
+        const value = all_traits.account[`clearbit/${clearbit_name}`];
+        if (value) {
+          _.set(all_traits.account, top_level_name, { value, operation: "setIfNull" });
+        }
+      });
+
+      const client = this.hull.asUser(ident);
+
+      client.traits(all_traits.user);
```

Comment:
we are not chaining those promises calls, let's pass that to the returned value to make sure we don't break the async chain.
one suggestion how to do it easily is to create array at the top of the function:
```js
const promises = [];
```
then push promises then: `promises.push(client.traits({ traits }))`
and return them: `return Promise.all(promises)`.